test/jasmine/tests/gl3d_plot_interact_test.js

```diff
@@ -568,6 +568,99 @@ describe('Test gl3d plots', function() {
         .then(done);
     });
 
+    it('@gl should set the camera projection type to perspective if the camera.projection.type is not set', function(done) {
+        Plotly.plot(gd, {
+            data: [{
+                type: 'scatter3d',
+                x: [1, 2, 3],
+                y: [2, 3, 1],
+                z: [3, 1, 2]
+            }],
+            layout: {
+                scene: {
+                    camera: {
+                    }
+                }
+            }
+        })
+        .then(delay(20))
+        .then(function() {
+            expect(gd._fullLayout.scene.camera.projection.type === 'perspective').toBe(true);
+        })
+        .then(function() {
+            expect(gd._fullLayout.scene._scene.glplot.camera._ortho).toBe(false);
+        })
+        .then(done);
+    });
+
+    it('@gl should set the camera projection type to orthographic if the camera.projection.type is set to orthographic', function(done) {
+        Plotly.plot(gd, {
+            data: [{
+                type: 'scatter3d',
+                x: [1, 2, 3],
+                y: [2, 3, 1],
+                z: [3, 1, 2]
+            }],
+            layout: {
+                scene: {
+                    camera: {
+                        projection: {
+                            type: 'orthographic'
+                        }
+                    }
+                }
+            }
+        })
+        .then(delay(20))
+        .then(function() {
+            expect(gd._fullLayout.scene.camera.projection.type === 'orthographic').toBe(true);
+        })
+        .then(function() {
+            expect(gd._fullLayout.scene._scene.glplot.camera._ortho).toBe(true);
+        })
+        .then(done);
+    });
+
+    it('@gl should enable orthographic & perspective projections using relayout', function(done) {
+        Plotly.plot(gd, {
+            data: [{
+                type: 'scatter3d',
+                x: [1, 2, 3],
+                y: [2, 3, 1],
+                z: [3, 1, 2]
+            }],
+            layout: {
+                scene: {
+                    camera: {
+                        projection: {
+                            type: 'perspective'
+                        }
+                    }
+                }
+            }
+        })
+        .then(delay(20))
+        .then(function() {
+            return Plotly.relayout(gd, 'scene.camera.projection.type', 'orthographic');
+        })
+        .then(function() {
+            expect(gd._fullLayout.scene.camera.projection.type === 'orthographic').toBe(true);
+        })
+        .then(function() {
+            expect(gd._fullLayout.scene._scene.glplot.camera._ortho).toBe(true);
+        })
+        .then(function() {
+            return Plotly.relayout(gd, 'scene.camera.projection.type', 'perspective');
```

Comment:
@archmoj could we add a drag test for orthographic projections to (at least) make sure the relayout callbacks are triggered properly?

To do so, we could a `relayout(gd, 'scene.projection.type', 'orthographic')` and one or two `expect()`s to this test case:

https://github.com/plotly/plotly.js/blob/79fa245cc5ce6e490ae33211f2de2e5a8b8027bb/test/jasmine/tests/gl3d_plot_interact_test.js#L1117-L1215

Reply:
@etpinard Thanks for your review. In https://github.com/plotly/plotly.js/pull/3550/commits/74f839bdec46631f5e9b5e69e26905ccb8091627 I added similar tests for `orthographic` projection to cover interactions and plots with multiple scenes. New `gl-plot3d` and `gl-axes3d` modules are published and applied in https://github.com/plotly/plotly.js/pull/3550/commits/9af35cc2f253ef62cbed176ea434d00824b3afac.

Reply:
Beautifully done. Thanks!